diff --git a/libraries/node/deps/v8/tools/js2c.py b/libraries/node/deps/v8/tools/js2c.py
index b5436f9..2ae93cd 100755
--- a/libraries/node/deps/v8/tools/js2c.py
+++ b/libraries/node/deps/v8/tools/js2c.py
@@ -499,7 +499,7 @@ def BuildMetadata(sources, source_bytes, native_type):
     "get_index_cases": "".join(get_index_cases),
     "get_script_source_cases": "".join(get_script_source_cases),
     "get_script_name_cases": "".join(get_script_name_cases),
-    "type": native_type,
+    "type": native_type.replace('..\..\..\..\..\..\..\..\libraries\\node\\deps\\v8\\tools\\gyp\\',''),
   }
   return metadata

