diff --git a/libraries/node/deps/v8/build/toolchain.gypi b/libraries/node/deps/v8/build/toolchain.gypi
index 1d47360..77f13cf 100644
--- a/libraries/node/deps/v8/build/toolchain.gypi
+++ b/libraries/node/deps/v8/build/toolchain.gypi
@@ -659,7 +659,7 @@
               }],
               ['v8_target_arch=="x64"', {
                 # TODO(2207): remove this option once the bug is fixed.
-                'WholeProgramOptimization': 'true',
+                # 'WholeProgramOptimization': 'true',
               }],
             ],
           },
@@ -832,7 +832,7 @@
                   }],
                   ['v8_target_arch=="x64"', {
                     # TODO(2207): remove this option once the bug is fixed.
-                    'WholeProgramOptimization': 'true',
+                    # 'WholeProgramOptimization': 'true',
                   }],
                 ],
               },

diff --git a/deps/v8/tools/js2c.py b/deps/v8/tools/js2c.py
index 77485f6..088d09a
--- a/libraries/node/deps/v8/tools/js2c.py
+++ b/libraries/node/deps/v8/tools/js2c.py
@@ -571,6 +571,9 @@ def main():
       sources.js: JS internal sources or macros.py.""")
   (options, args) = parser.parse_args()
 
+  args[1] = args[1].replace('../','').replace('..\\','').replace('libraries\\','').replace('node\\','').replace('deps\\','').replace('v8\\','').replace('tools\\','').replace('gyp\\','')
+  args[2] = args[2].replace('../','').replace('..\\','').replace('libraries\\','').replace('node\\','').replace('deps\\','').replace('v8\\','').replace('tools\\','').replace('gyp\\','')
+
   JS2C(args[3:], args[0], args[1], args[2], options.raw, options.startup_blob)


diff --git a/libraries/node/src/node.js b/libraries/node/src/node.js
index 5046038..d401978 100644
--- a/libraries/node/src/node.js
+++ b/libraries/node/src/node.js
@@ -482,8 +482,14 @@
         break;
 
       default:
-        // Probably an error on in uv_guess_handle()
-        throw new Error('Implement me. Unknown stream file type!');
+        if(process.platform === 'win32' && (fd === 1 || fd === 2)) {
+          var fs = NativeModule.require('fs');
+          stream = new fs.SyncWriteStream(fd);
+          stream._type = 'fs'
+        } else {
+          // Probably an error on in uv_guess_handle()
+          throw new Error('Implement me. Unknown stdin file type!');
+        }
     }
 
     // For supporting legacy API we put the FD here.
@@ -554,8 +560,13 @@
           break;
 
         default:
-          // Probably an error on in uv_guess_handle()
-          throw new Error('Implement me. Unknown stdin file type!');
+          if(process.platform === 'win32' && (fd === 0)) {
+            var fs = NativeModule.require('fs');
+            stdin = new fs.ReadStream(null, { fd: fd });
+          } else {
+            // Probably an error on in uv_guess_handle()
+            throw new Error('Implement me. Unknown stdin file type!');
+          }
       }
 
       // For supporting legacy API we put the FD here.
@@ -564,7 +575,7 @@
       // stdin starts out life in a paused state, but node doesn't
       // know yet.  Explicitly to readStop() it to put it in the
       // not-reading state.
-      if (stdin._handle && stdin._handle.readStop) {
+      if (stdin._handle && stdin._handle.readStop && process.platform != 'win32') {
         stdin._handle.reading = false;
         stdin._readableState.reading = false;
         stdin._handle.readStop();
