diff --git a/deps/v8/build/toolchain.gypi b/deps/v8/build/toolchain.gypi
index 1d47360..a99f1b0 100644
--- a/deps/v8/build/toolchain.gypi
+++ b/deps/v8/build/toolchain.gypi
@@ -659,7 +659,7 @@
               }],
               ['v8_target_arch=="x64"', {
                 # TODO(2207): remove this option once the bug is fixed.
-                'WholeProgramOptimization': 'true',
+                # 'WholeProgramOptimization': 'true',
               }],
             ],
           },
@@ -824,17 +824,17 @@
                 'EnableIntrinsicFunctions': 'true',
                 'FavorSizeOrSpeed': '0',
                 'StringPooling': 'true',
-                'conditions': [
-                  ['component=="shared_library"', {
+                # 'conditions': [
+                  # ['component=="shared_library"', {
                     'RuntimeLibrary': '2',  #/MD
-                  }, {
-                    'RuntimeLibrary': '0',  #/MT
-                  }],
-                  ['v8_target_arch=="x64"', {
+                  # }, {
+                  #  'RuntimeLibrary': '0',  #/MT
+                  # }],
+                  # ['v8_target_arch=="x64"', {
                     # TODO(2207): remove this option once the bug is fixed.
-                    'WholeProgramOptimization': 'true',
-                  }],
-                ],
+                    # 'WholeProgramOptimization': 'true',
+                  # }],
+                # ],
               },
               'VCLinkerTool': {
                 'LinkIncremental': '1',
diff --git a/libraries/node/src/node.js b/libraries/node/src/node.js
index e5833cb..6263c11 100644
--- a/libraries/node/src/node.js
+++ b/libraries/node/src/node.js
@@ -592,9 +592,16 @@
         }
         break;
 
-      default:
-        // Probably an error on in uv_guess_handle()
-        throw new Error('Implement me. Unknown stream file type!');
+      default:
+        // See https://groups.google.com/forum/#!topic/nodejs/t1V8psq309w
+        if(process.platform === 'win32' && (fd === 1 || fd === 2)) {
+          var fs = NativeModule.require('fs');
+          stream = new fs.SyncWriteStream(fd);
+          stream._type = 'fs';
+        } else {
+          // Probably an error on in uv_guess_handle()
+          throw new Error('Implement me. Unknown stream file type!');
+        }
     }
 
     // For supporting legacy API we put the FD here.
@@ -664,9 +671,15 @@
           });
           break;
 
-        default:
-          // Probably an error on in uv_guess_handle()
-          throw new Error('Implement me. Unknown stdin file type!');
+        default:
+          // See https://groups.google.com/forum/#!topic/nodejs/t1V8psq309w
+          if(process.platform === 'win32' && fd === 0) {
+            var fs = NativeModule.require('fs');
+            stdin = new fs.ReadStream(null, { fd: fd });
+          } else {
+            // Probably an error on in uv_guess_handle()
+            throw new Error('Implement me. Unknown stdin file type!');
+          }
       }
 
       // For supporting legacy API we put the FD here.
